arch/arm/src/c5471/c5471_irq.c

```diff
@@ -59,7 +59,7 @@ volatile uint32_t *g_current_regs[1];
  * because we know that correct IRAM area is 0xffc00000.
  */
 
-extern int _svectors; /* Type does not matter */
+extern int *_svectors; /* Type does not matter */
```

Comment:
```suggestion
extern up_vector_t _svectors; /* Type does not matter */
```
and then can change `up_vector_t *dest = (up_vector_t *)&_svectors;` to `up_vector_t *dest = &_svectors;`

Reply:
if _svectors is changed to:
extern up_vector_t _svectors;
compiler will assume _svectors has only one element, and report the out of range:
```
c5471/c5471_irq.c: Fix error: '__builtin_memcpy' forming offset [4, 31] is out of the bounds [0, 4] of object '_svectors' with type 'int'
```

Reply:
Anyway, I change the type to void * which is better than int *.

Reply:
but even we declare `extern up_vector_t _svectors[];`, we still need add & and cast to _svectors.

Reply:
No the `&` is not needed for arrays

Reply:
For arrays the `if (a == &a)` always evaluated to `true`

Reply:
Ok, change to `extern up_vector_t _svectors[]; `